misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -814,10 +815,39 @@ public ExpressionStatementNode transform(ExpressionStatementNode expressionState
     @Override
     public CheckExpressionNode transform(CheckExpressionNode checkExpressionNode) {
         Token checkKeyword = formatToken(checkExpressionNode.checkKeyword(), 1, 0);
-        ExpressionNode expressionNode = formatNode(checkExpressionNode.expression(), env.trailingWS, env.trailingNL);
+        ExpressionNode expressionNode;
+        if (checkExpressionNode.onFailCheck().isPresent()) {
+            expressionNode = formatNode(checkExpressionNode.expression(), 1, 0);
+        } else {
+            expressionNode = formatNode(checkExpressionNode.expression(), env.trailingWS, env.trailingNL);
+            return checkExpressionNode.modify()
+                    .withCheckKeyword(checkKeyword)
+                    .withExpression(expressionNode)
+                    .apply();
+        }
+
+        OnFailCheckNode onFailCheckNode = formatNode(checkExpressionNode.onFailCheck().get(),  0, 0);
```

Comment:
Single return preferred.
```suggestion
        OnFailCheckNode onFailCheckNode  = null;
        if (checkExpressionNode.onFailCheck().isPresent()) {
            expressionNode = formatNode(checkExpressionNode.expression(), 1, 0);
            onFailCheckNode = formatNode(checkExpressionNode.onFailCheck().get(),  0, 0);
        } else {
            expressionNode = formatNode(checkExpressionNode.expression(), env.trailingWS, env.trailingNL);
        }

```